packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1476,6 +1509,22 @@ class TextInput {
       case 'TextInputClient.updateEditingState':
         _currentConnection!._client.updateEditingValue(TextEditingValue.fromJSON(args[1] as Map<String, dynamic>));
         break;
+      case 'TextInputClient.updateEditingStateWithDeltas':
+        final List<TextEditingDelta> batchDeltas = <TextEditingDelta>[];
+
+        final Map<String, dynamic> encoded = args[1] as Map<String, dynamic>;
+
+        // On Android there could be a situation where multiple edits done to
+        // an editing state are accumulated before being sent to the framework.
+        // This is called batch editing on Android, and on other platforms the
+        // TextInputPlugin does not adhere to this type of editing.
+        for (final dynamic encodedDelta in encoded['batchDeltas']) {
```

Comment:
Looks good always using an array of batchDeltas now 👍

Reply:
Actually not to overly complicate things, but should we just call this `deltas` now?

Reply:
I agree will also remove the comment above this regarding batch editing.